src/lib.rs

```diff
@@ -143,102 +143,18 @@
 //! ```
 //!
 #![warn(missing_docs, missing_debug_implementations, unreachable_pub)]
+#![cfg_attr(not(test), no_std)]
 
+#[cfg(feature = "alloc")]
+extern crate alloc;
+
+#[cfg(feature = "alloc")]
+pub mod arc_waker;
```

Comment:
This causes `cargo check` and `cargo clippy` to emit a warning. Unfortunately I forgot to make warnings a hard failure in the CI, my bad...
In any case, since the content is re-exported in the main namespace, I'd rather keep this module private.